updater/lib/dependabot/job.rb

```diff
@@ -43,87 +45,132 @@ class Job
       dependency_groups
       dependency_group_to_refresh
       repo_private
-    ).freeze
-
-    attr_reader :allowed_updates,
-                :credentials,
-                :dependencies,
-                :existing_pull_requests,
-                :existing_group_pull_requests,
-                :id,
-                :ignore_conditions,
-                :package_manager,
-                :requirements_update_strategy,
-                :security_advisories,
-                :security_updates_only,
-                :source,
-                :token,
-                :vendor_dependencies,
-                :dependency_groups,
-                :dependency_group_to_refresh
+    ).freeze, T::Array[Symbol])
+
+    sig { returns(T::Array[T::Hash[String, T.untyped]]) }
+    attr_reader :allowed_updates
+
+    sig { returns(T::Array[T.untyped]) }
+    attr_reader :credentials
+
+    sig { returns(T.nilable(T::Array[String])) }
+    attr_reader :dependencies
+
+    sig { returns(T::Array[T.untyped]) }
+    attr_reader :existing_pull_requests
+
+    sig { returns(T::Array[T.untyped]) }
+    attr_reader :existing_group_pull_requests
+
+    sig { returns(T.any(Integer, String)) }
+    attr_reader :id
+
+    sig { returns(T::Array[T.untyped]) }
+    attr_reader :ignore_conditions
+
+    sig { returns(String) }
+    attr_reader :package_manager
+
+    sig { returns(T.nilable(String)) }
+    attr_reader :requirements_update_strategy
+
+    sig { returns(T::Array[T.untyped]) }
```

Comment:
```suggestion
    sig { returns(T::Array[Dependabot::SecurityAdvisory]) }
```

Reply:
Same as below, it's not constructed into a class yet.